### PR TITLE
feat(board):Update M5Stack related board and pin configuration

### PR DESCRIPTION
**Pull request by Tinyu-Zhao**
Update M5Stack related board and pin configuration to fix some of the board issues.

### CODE REVIEW

File: variants/m5stack_atom/pins_arduino.h [hunk 1]
```diff
@@ -3,6 +3,14 @@
...
 #include <stdint.h>
 
+#define EXTERNAL_NUM_INTERRUPTS 16
+#define NUM_DIGITAL_PINS        40
+#define NUM_ANALOG_INPUTS       16
+
+#define analogInputToDigitalPin(p) (((p) < 20) ? (esp32_adc2gpio[(p)]) : -1)
+#define digitalPinToInterrupt(p)   (((p) < 40) ? (p) : -1)
+#define digitalPinHasPWM(p)        (p < 34)
+
```
Comment (P-R-O-C-H-Y) [inline variants/m5stack_atom/pins_arduino.h:13]: Hi @Tinyu-Zhao, those defines are in 3.X version defined globally per SoC in Arduino.h file.
Defining them here will trigger redefinition error/warning. Can you please remove those from the PR? Thanks
```suggestion
```

Comment (Tinyu-Zhao) [reply variants/m5stack_atom/pins_arduino.h:13]: I would like to ask, the constants defined in Arduino.h will not be added again in variants, right?

image

Comment (P-R-O-C-H-Y) [reply variants/m5stack_atom/pins_arduino.h:13]: No they won't :)

Comment (Tinyu-Zhao) [reply variants/m5stack_atom/pins_arduino.h]: > No they won't :)

PTAL🙏

File: boards.txt [hunk 2]
```diff
@@ -18494,15 +18494,9 @@ m5stack_core.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
...
+m5stack_core.menu.PartitionScheme.rainmaker=RainMaker
 m5stack_core.menu.PartitionScheme.rainmaker.build.partitions=rainmaker
-m5stack_core.menu.PartitionScheme.rainmaker.upload.maximum_size=1966080
-m5stack_core.menu.PartitionScheme.rainmaker_4MB=RainMaker 4MB No OTA
-m5stack_core.menu.PartitionScheme.rainmaker_4MB.build.partitions=rainmaker_4MB_no_ota
-m5stack_core.menu.PartitionScheme.rainmaker_4MB.upload.maximum_size=4038656
-m5stack_core.menu.PartitionScheme.rainmaker_8MB=RainMaker 8MB
-m5stack_core.menu.PartitionScheme.rainmaker_8MB.build.partitions=rainmaker_8MB
-m5stack_core.menu.PartitionScheme.rainmaker_8MB.upload.maximum_size=4116480
+m5stack_core.menu.PartitionScheme.rainmaker.upload.maximum_size=3145728
```
Comment (P-R-O-C-H-Y) [inline boards.txt:18499]: @Tinyu-Zhao Why are you rollbacking the RainMaker partitions? It got fixed recently in 3.0.4.
As RainMaker partition uses 2 OTA partitions, the maximum upload size should be 1966080, not 3145728.
The binary will flash fine, but the board will stay in boot loop.

Comment (Tinyu-Zhao) [reply boards.txt]: Thanks for the tip, I just pushed the update.

**Review by P-R-O-C-H-Y — APPROVED** [review]
LGTM

**Review by lucasssvaz — APPROVED** [review]

File: boards.txt [hunk 3]
```diff
@@ -19913,6 +19911,8 @@ m5stack_atom.menu.EraseFlash.all.upload.erase_cmd=-e
 ##############################################################
 
 m5stack_atoms3.name=M5AtomS3
+m5stack_atoms3.vid.0=0x303a
+m5stack_atoms3.pid.0=0x1001
```
Comment (me-no-dev) [inline boards.txt:19915]: these are non-unique PID and VID. Please remove them from all boards!

Comment (me-no-dev) [reply boards.txt:19915]: @P-R-O-C-H-Y please make sure that such things do not pass!!!

Comment (P-R-O-C-H-Y) [reply boards.txt:19915]: yep, missed that.

Comment (Tinyu-Zhao) [reply boards.txt:19915]: But I see that many boards have added it. Is it enough to just use a different PID and VID?

image

Comment (me-no-dev) [reply boards.txt]: the boards that use different PID and VID, use it through the OTG port (TinyUSB). The ones you had are for the integrated CDC/JTAG port and should only be used/added to the "ESP32 Family Board" Else there are issues with the Arduino IDE and detecting boards.

**Review by me-no-dev — APPROVED** [review]

File: boards.txt (hunk 3 above)
Comment (Tinyu-Zhao) [reply boards.txt]: Thank you for your reply.

**Pull request merged**